enum.rb

```diff
@@ -0,0 +1,118 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    index = 0
+    array = to_a
+    while index < array.length
+      yield(array[index])
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_wth_index) unless block_given?
+
+    length.times do |index|
+      yield(self[index], index)
+    end
+    self
+  end
+
+  def my_select
+    return to_enum if block_given? == false
+
+    array = []
+    my_each { |index| array.push(index) if yield(index) }
+    array
+  end
+
+  def my_all?(_arr = nil)
+    condition = true
+    if block_given?
+      my_each { |index| condition = false unless yield(index) }
+    else
+      my_each { |index| condition = false if index.nil? || index == false }
+    end
+    condition
+  end
+
+  def my_any?(arr = nil)
+    condition = false
+    my_each do |index|
+      if block_given?
+        condition = true if yield(index)
+      elsif arr.nil?
+        condition = true unless index
+      elsif arr === index
+        condition = true
+      end
+    end
+    condition
+  end
+
+  def my_none?(arr = nil)
+    condition = true
+    my_each do |index|
+      if block_given?
+        condition = false if yield index
+      elsif arr.nil?
+        condition = false if index
+      elsif arr === index
+        my_each { |_index| condition = false }
+      end
+    end
+    condition
+  end
```

Comment:
[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end
```
To make this work we have to use block.call(element) instead of yield(element) in my_any?